assets/components/stripePopUpButton/stripePopUpButton.jsx

```diff
@@ -4,40 +4,44 @@
 
 import React from 'react';
 import { SvgCreditCard } from 'components/svg/svg';
-import { connect } from 'react-redux';
+import type { Currency } from 'helpers/internationalisation/currency';
 
 import {
-  openStripeOverlay,
   setupStripeCheckout,
-} from 'helpers/stripeCheckout/stripeCheckoutActions';
-
+  openDialogBox,
+} from 'helpers/stripeCheckout/stripeCheckout';
 
 // ---- Types ----- //
 
 /* eslint-disable react/no-unused-prop-types */
 type PropTypes = {
-  stripeLoaded: boolean,
-  setupStripeCheckout: Function,
-  openStripeOverlay: Function,
-  stripeClick?: Function,
   amount: number,
-  email: string,
   callback: Function,
+  closeHandler?: Function,
+  currency: Currency,
+  email: string,
+  isTestUser: boolean,
+  isValid?: Function,
 };
 /* eslint-enable react/no-unused-prop-types */
 
 
+function isStripeSetup(): boolean {
+  return window.StripeCheckout !== undefined;
+}
+
+
 // ----- Component ----- //
 
 const StripePopUpButton = (props: PropTypes) => {
 
-  if (!props.stripeLoaded) {
-    props.setupStripeCheckout(props.callback);
+  if (!isStripeSetup()) {
+    setupStripeCheckout(props.callback, props.closeHandler, props.currency.iso, props.isTestUser);
   }
 
   const onClick = () => {
-    if (props.stripeClick) {
-      props.stripeClick(() => props.openStripeOverlay(props.amount, props.email));
+    if (props.isValid && props.isValid()) {
```

Comment:
Hmmm, reading `isValid` without knowing about the context in which Stripe is being imported makes it sound like you're checking to see if Stripe itself is valid (whereas presumably this refers to a form of some sort). Would something like `canOpen`, which assumes nothing about the external environment, work as an alternative?

Reply:
Fair enough, I will update it.

Reply:
Updated in 8725f03c1c119ee74e14b50fcc379fa3ffe2389d